presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryBuilder.java

```diff
@@ -14,38 +14,32 @@
 package com.facebook.presto.orc.writer;
 
 import com.facebook.presto.common.block.Block;
-import com.facebook.presto.common.block.BlockBuilder;
 import com.facebook.presto.common.block.VariableWidthBlockBuilder;
 import com.facebook.presto.orc.array.IntBigArray;
+import io.airlift.slice.Slice;
 import org.openjdk.jol.info.ClassLayout;
 
 import static com.facebook.presto.common.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
 import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Verify.verify;
 import static it.unimi.dsi.fastutil.HashCommon.arraySize;
 import static java.lang.Math.min;
 import static java.util.Objects.requireNonNull;
 
 // TODO this class is not memory efficient.  We can bypass all of the Presto type and block code
-// since we are only interested in a hash of byte arrays.  The only place an actual block is needed
-// is during conversion to direct, and in that case we can use a slice array block.  This code
-// can use store the data in multiple Slices to avoid a large contiguous allocation.
+// since we are only interested in a hash of byte arrays.
 public class DictionaryBuilder
```

Comment:
Given we are for sure specializing this class. Let's rename it to reflect it

Reply:
Some names I can think of are SliceDicitonary or SliceDictionaryBuilder. Please let me know if you prefer some other names.